### DISCUSSION TITLE
How to keep image format and overwrite existing file with vipsthumbnail?

### DISCUSSION THREAD

**Discussion by hydrajump** [libvips/libvips#2774] [Q&A] [answered]
If I use `vipsthumbnail` on a PNG file I've noticed that the output is JPEG. For example,

```
vipsthumbnail --vips-info --size 400x400 image.png
VIPS-INFO: 22:34:29.547: thumbnailing image.png
VIPS-INFO: 22:34:29.548: selected loader is VipsForeignLoadPngFile
VIPS-INFO: 22:34:29.548: input size is 200 x 200
VIPS-INFO: 22:34:29.548: loading with factor 1 pre-shrink
VIPS-INFO: 22:34:29.549: pre-shrunk size is 200 x 200
VIPS-INFO: 22:34:29.549: converting to processing space srgb
VIPS-INFO: 22:34:29.549: residual scale 2 x 2
VIPS-INFO: 22:34:29.550: thumbnailing image.png as ./tn_image.jpg
```

Is it possible to keep the original image format?

Also, when I attempt to overwrite the existing file with the transformed output I get the following error:

```
vipsthumbnail --size 400x400 image.jpg --output image.jpg

(vipsthumbnail:62459): VIPS-WARNING **: 22:26:50.743: error in tile 0 x 0

(vipsthumbnail:62459): VIPS-WARNING **: 22:26:50.743: error in tile 0 x 10
vipsthumbnail: unable to thumbnail image.jpg
VipsJpeg: Premature end of JPEG file
VipsJpeg: Not a JPEG file: starts with 0xff 0xd9

(vipsthumbnail:62459): VIPS-WARNING **: 22:26:50.744: read gave 1 warnings

(vipsthumbnail:62459): VIPS-WARNING **: 22:26:50.744: 
```

The file gets overwritten  with 0 bytes 😢 

Is shell magic, e.g. `mv tn_image.jpg image.jpg`  the only way to overwrite the original file?

Thanks.

**Comment by jcupitt** [comment, marked as answer]
Hi @hydrajump,

Sure, try:

```
vipsthumbnail wtc.png --size 400x400 --output x.png
```

To write `x.png`.

vipsthumbnail streams images, so it'll read the input while it writes the output. This is why memory use it low -- it never loads the whole input image. The downside is you can't have the input and the output in the same file. 

Yes, you need to output to a temporary file, then move that back over the input.
